scipy/linalg/_decomp_cholesky.py

```diff
@@ -374,3 +376,105 @@ def cho_solve_banded(cb_and_lower, b, overwrite_b=False, check_finite=True):
         raise ValueError('illegal value in %dth argument of internal pbtrs'
                          % -info)
     return x
+
+
+def cholesky_update(R, z, downdate=False, lower=False, overwrite_R=False,
+               overwrite_z=False, eps=None, check_finite=True):
+    """
+    Performs a rank-1 update or downdate of Cholesky factors.
+
+    Cholesky rank-1 update and downdate algorithm taken from [1]_ such that
+    an update results in ``D'D = R'R + zz'`` and a downdate results in ``D'D =
+    R'R - zz'``
+
+    Parameters
+    ----------
+    R : (N, N) array_like
+        The 2D input data from which the triangular part will be used to
+        read the Cholesky factor.The remaining parts are ignored.
+    z : (N,) array_like
+        A 1D update/downdate vector
+    downdate: bool, optional
+        The type of rank-1 modification desired. False indicates an update
+        while True indicates a downdate. Default is False.
+    lower: bool, optional
+        Whether input array is upper or lower triangular Cholesky
+        factorization. Default is upper-triangular.
+    overwrite_R: bool, optional
+        If set to True the entries of the Cholesky factor array will be
+        modified during the computations instead of creating a new array.
+        Default is False.
+    overwrite_z : bool, optional
+        If set to True the entries of the update array will be modified during
+        the computations instead of creating a new array. Default is False.
+    eps : float, optional
```

Comment:
```suggestion
    eps : non-negative float, optional
```
Probably need to check for non-negativity.